src/compas/geometry/primitives/polyline.py

```diff
@@ -413,3 +413,68 @@ def divide_polyline_by_length(self, length, strict=True, tol=1e-06):
         return division_pts
 
     divide_by_length = divide_polyline_by_length
+
+    def extend(self, length):
+        """Extends a polyline by a given length, by moving the first and/or last point tangentially.
+
+        Parameters:
+        -----------
+        length: float or tuple[float, float]
+            end_extension or tuple[start_extension, end_extension]
+
+        Returns
+        -------
+        :class:`~compas.geometry.Polyline`
+
+        """
+        try:
+            start, end = length
+            self.points[0] = self.points[0] + self.lines[0].vector.unitized().scaled(-start)
+        except TypeError:
+            start = end = length
+        self.points[-1] = self.points[-1] + self.lines[-1].vector.unitized().scaled(end)
```

Comment:
I think you could move this line down (outside the `try` block), right?
```suggestion
        except TypeError:
            start = end = length
        self.points[0] = self.points[0] + self.lines[0].vector.unitized().scaled(-start)
        self.points[-1] = self.points[-1] + self.lines[-1].vector.unitized().scaled(end)
```

Reply:
I kept in inside on purpose. My reasoning being: The standard behavior of the function is to only extend the line from the end this was also taking Tom's comments into account. So by keeping it within the `try` gives us option to also extend from the beginning. In the rare case that you would only like to extend from the beginning you can simply input `(length, 0)` into the function. Extending both start and end `(length0, length1)` and extending just the end would be either `length` or `(0, length)